src/runtime_src/core/common/info_platform.cpp

```diff
@@ -156,8 +156,28 @@ add_controller_info(const xrt_core::device* device, ptree_type& pt)
                           % ((versionValue >> (1 * 8)) & 0xFF) // Minor
                           % ((versionValue >> (0 * 8)) & 0xFF)); // Version
     cmc.add("version", version);
-    cmc.add("serial_number", xrt_core::device_query<xq::xmc_serial_num>(device));
-    cmc.add("oem_id", xq::oem_id::parse(xrt_core::device_query<xq::oem_id>(device)));
+    std::string sn = xrt_core::device_query<xq::xmc_serial_num>(device);
+    if (sn.empty()) {
+      try {
+        sn = xrt_core::device_query<xq::hwmon_sdm_serial_num>(device);
+      }
+      catch (const xq::no_such_key&) {
+        // Ignoring if not available
+      }
+    }
+    cmc.add("serial_number", sn);
+
+    std::string oid = xq::oem_id::parse(xrt_core::device_query<xq::oem_id>(device));
+    if (oid.empty()) {
```

Comment:
Hi @rozumx & @AShivangi  , oem_id::parse() function returns "N/A" if it is unable to find oem_id in the list. So, this function returns string "N/A" on versal. Hence, either add new cond for compare oid with "N/A" also or modify parse function to return empty string instead of "N/A". Please suggest. Thank you.

Reply:
The correct behavior should be empty string if there is no serial number.   
Being that we are running out of room regarding the release, please do the following:

1. Add the following code after line 179:
```
if (boost::iequal(oid, "N/A")
   old.clear();
```
2. Create a CR for 2022.2 requesting that oem_id::parse() function return back an empty string.

Doing so will allow this code to implement the correct behavior and give us time to address the correct overall behavior post 2022.1.

Reply:
I have added your suggested code